### ISSUE TITLE
[v2][gatsby-plugin-react-css-modules] Minify class names?

### ISSUE THREAD

**Issue by sktt** [gatsbyjs/gatsby#8153]
## Summary

css-module class names can be made shorter in a production build.

### Basic example

We currently always use the template `[name]--[local]--[hash:base64:5]` for both development and production.

My suggestion is to shorten this in production builds, eg. 
```diff
+  const isDev = process.env.NODE_ENV !== `production`
  actions.setBabelPlugin({
    name: `babel-plugin-react-css-modules`,
    options: {
-      generateScopedName: `[name]--[local]--[hash:base64:5]`,
+      generateScopedName: isDev ? `[name]--[local]--[hash:base64:5]` : `[hash:base64:5]`,
-      webpackHotModuleReloading: process.env.NODE_ENV !== `production`
+      webpackHotModuleReloading: isDev
    }
  });
```

### Motivation

It would make the build a byte or two smaller.. Probably gzip already cut most the bytes but still...

**Comment by kakadiadarpan** [comment]
@sktt How about we give everyone a control over this through options in `gatsby-plugin-react-css-modules`?

```js
// In your gatsby-config.js
plugins: [
  {
    resolve: `gatsby-plugin-react-css-modules`,
    options: {
      minifyClassNames: true,
      // *.css files are included by default.
      // To support another syntax (e.g. SCSS),
      // add `postcss-scss` to your project's devDependencies
      // and add the following option here:
      filetypes: {
        ".scss": { syntax: `postcss-scss` },
      },

      // Exclude global styles from the plugin using a RegExp:
      exclude: `\/global\/`,
    },
  },
]
```

Read the value of `minifyClassNames` and accordingly decide whether to minify it for production builds or not.

**Comment by stefanprobst** [comment]
This is what the built-in css-modules handling does (with a helpful link):
https://github.com/gatsbyjs/gatsby/blob/4be5cfdf5d6925f26f914518dfc51d53f979cd6d/packages/gatsby/src/utils/webpack-utils.js#L197-L198

**Comment by sktt** [comment]
I am happy with https://github.com/gatsbyjs/gatsby/pull/8038 it will do perfectly

**Comment by kakadiadarpan** [comment]
> @sktt How about we give everyone a control over this through options in `gatsby-plugin-react-css-modules`?
> 
> ```js
> // In your gatsby-config.js
> plugins: [
>   {
>     resolve: `gatsby-plugin-react-css-modules`,
>     options: {
>       minifyClassNames: true,
>       // *.css files are included by default.
>       // To support another syntax (e.g. SCSS),
>       // add `postcss-scss` to your project's devDependencies
>       // and add the following option here:
>       filetypes: {
>         ".scss": { syntax: `postcss-scss` },
>       },
> 
>       // Exclude global styles from the plugin using a RegExp:
>       exclude: `\/global\/`,
>     },
>   },
> ]
> ```
> 
> Read the value of `minifyClassNames` and accordingly decide whether to minify it for production builds or not.

@sktt #8038 only enables the passing down of options to the plugin. It doesn't include the changes we discussed above. Are you sure you want to close this issue?

**Comment by sktt** [comment]
Yes, #8038 is better imho, it does not intoduce extra configuration so we can just rely on the docs for the babel plugin!

**Comment by sktt** [comment]
However, this will not work. see @stefanprobst comment. The class name generation is also hardcoded in there.. i guess thats the reason..

**Comment by kakadiadarpan** [comment]
@sktt that's the reason why I asked [above](https://github.com/gatsbyjs/gatsby/issues/8153#issuecomment-423423548) if you were sure about closing it.

Would you like to submit a PR for this? You'll need to update this file:
https://github.com/gatsbyjs/gatsby/blob/4be5cfdf5d6925f26f914518dfc51d53f979cd6d/packages/gatsby/src/utils/webpack-utils.js#L197-L198

And also the documentation for the same package.

**Comment by sktt** [comment]
could look at it in the weekend. But I haven't done any webpack tweaking for gatsby so far so to me it's not obvious what needs to be done here :)

**Comment by vijaykolar** [comment]
Hi,

Hash classes are generating in local but not in production
